src/builtin.c

```diff
@@ -657,6 +657,48 @@ static jv f_format(jq_state *jq, jv input, jv fmt) {
     }
     jv_free(input);
     return line;
+  } else if (!strcmp(fmt_s, "urid")) {
+    jv_free(fmt);
+    input = f_tostring(jq, input);
+
+    jv line = jv_string("");
+    const char *errmsg =  "is not a valid uri encoding";
+    char *s = (char *)jv_string_value(input);
+    while (*s) {
+      if (*s != '%') {
+        line = jv_string_append_buf(line, s++, 1);
+      } else {
+        unsigned char unicode[4] = {0};
+        int b = 0;
+        // check leading bits of first octet to determine length of unicode character
+        // (https://datatracker.ietf.org/doc/html/rfc3629#section-3)
+        while (b == 0 || (b < 4 && unicode[0] >> 7 & 1 && unicode[0] >> (7-b) & 1)) {
+          if (*(s++) != '%') {
+            jv_free(line);
+            return type_error(input, errmsg);
+          }
+          for (int i=0; i<2; i++) {
+            unicode[b] <<= 4;
```

Comment:
Trying to follow, this will be a "nop" the first of the two iterations?

Reply:
yeah, same idea as this code [here](https://github.com/jqlang/jq/blob/137018d2ae6561a47fe8f118245acd467d7bd88a/src/jv_parse.c#L431-L444) (couldn't import it tho

Reply:
i see, would need a move it somewhere and have a new *_private.h (like jv_private.h) file to share the function guess, messy :( maybe easiest to keep it like this for now

Reply:
lol yeah i tried moving things around a bit to avoid code duplication but i didn't want to make any brash decisions so i went with this to keep it short and sweet :)